hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -1659,6 +1662,43 @@ public RegionLoad createRegionLoad(final String encodedRegionName) throws IOExce
     return r != null ? createRegionLoad(r, null, null) : null;
   }
 
+  public Map<String, Integer> uncacheStaleBlocks() {
+    Map<String, Pair<String, Long>> fullyCachedFiles =
+      this.getBlockCache().flatMap(BlockCache::getFullyCachedFiles).orElse(Collections.emptyMap());
+    Map<String, Integer> evictedFilesWithStaleBlocks = new ConcurrentHashMap<>();
+
+    ExecutorService executor = Executors.newFixedThreadPool(6);
```

Comment:
The threadpool should be global for HRegionServer. We don't want a new threadpool created on every client call to this method, that would potentially kill the RS if an operator calls it multiple times on a short period.